mackerel-plugin-windows-process-stats/lib/process-stats_windows.go

```diff
@@ -0,0 +1,117 @@
+// +build windows
+
+package mpwindowsprocessstats
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"sort"
+	"strconv"
+
+	"github.com/StackExchange/wmi"
+	mp "github.com/mackerelio/go-mackerel-plugin-helper"
+	"github.com/mackerelio/golib/logging"
+)
+
+var logger = logging.GetLogger("metrics.plugin.windows-process-stats")
+
+type Win32_PerfFormattedData_PerfProc_Process struct {
+	ElapsedTime          uint64
+	Name                 string
+	IDProcess            uint32
+	PercentProcessorTime uint64
+	WorkingSet           uint64
+}
+
+type WindowsProcessStatsPlugin struct {
+	Process string
+	Prefix  string
+}
+
+func getProcesses(processName string) ([]Win32_PerfFormattedData_PerfProc_Process, error) {
+	var procs []Win32_PerfFormattedData_PerfProc_Process
+
+	q := wmi.CreateQuery(&procs, "WHERE (Name like '"+processName+"' OR Name like '"+processName+"#%')")
+	if err := wmi.Query(q, &procs); err != nil {
+		return procs, err
+	}
+
+	sort.Slice(procs, func(i, j int) bool {
+		return procs[i].IDProcess < procs[j].IDProcess
+	})
+	return procs, nil
+}
+
+// FetchMetrics interface for mackerelplugin
+func (m WindowsProcessStatsPlugin) FetchMetrics() (map[string]interface{}, error) {
+	procs, err := getProcesses(m.Process)
+	if err != nil {
+		return nil, err
+	}
+	stat := make(map[string]interface{})
+	prefix := m.MetricKeyPrefix()
+	var re = regexp.MustCompile(`#[0-9]+$`)
+	for k, v := range procs {
+		name := re.ReplaceAllString(v.Name, "")
+		processName := name + "_" + strconv.Itoa(k)
+		metricNameCPU := prefix + "-windows-process-stats.cpu." + processName + ".percent_processor_time"
+		metricNameMemory := prefix + "-windows-process-stats.memory." + processName + ".working_set"
+		stat[metricNameCPU] = v.PercentProcessorTime
+		stat[metricNameMemory] = v.WorkingSet
+	}
+
+	return stat, nil
+}
+
+// GraphDefinition interface for mackerelplugin
+func (m WindowsProcessStatsPlugin) GraphDefinition() map[string](mp.Graphs) {
+	prefix := m.MetricKeyPrefix()
+	return map[string](mp.Graphs){
+		fmt.Sprintf("%s-windows-process-stats.cpu.#", prefix): mp.Graphs{
```

Comment:
The graph definition key should be`cpu.#`.
go-mackerel-plugin-helper will add `MetricKeyPrefix` to the result of `GraphDefinition()`.
ref. https://github.com/mackerelio/go-mackerel-plugin-helper/blob/master/mackerel-plugin.go#L359-L365